DataCleaningScripts/update_ndvi.R

```diff
@@ -0,0 +1,165 @@
+  center <- sf::st_sfc(sf::st_point(centroid),crs="WGS84")
+  #transform to NAD83(NSRS2007)/California Albers
+  center_transform <- sf::st_as_sf(center) %>% sf::st_transform(3488) 
+  portal_area_transform <- as(sf::st_buffer(center_transform, 1000), 'Spatial')
+  portal_area <- sp::spTransform(portal_area_transform, sp::CRS("+proj=utm +zone=12 
+                     +datum=WGS84 +units=m +no_defs +ellps=WGS84 +towgs84=0,0,0 "))
+  return(portal_area)
+  
+}
+
+
+#' Fetch new ndvi and put in new data order via getSpatialData
+#' 
+#' @param mindate minimum date to collect
+#' @param maxdate maximum date to collect
+#' @param targetpath path for data downloads 
+#'
+
+new_records <- function(mindate, maxdate, targetpath = tempdir()) {
+
+getSpatialData::set_archive(targetpath)
+getSpatialData::set_aoi(create_portal_area())
+getSpatialData::login_USGS(username = "weecology", password = Sys.getenv("USGS_PASSWORD"))
+
+records <- getSpatialData::get_records(time_range = c(mindate, maxdate),
+                                       products = "LANDSAT_8_C1")
+records <- records[records$level == "sr_ndvi",]
+records <- getSpatialData::check_availability(records)
+records <- getSpatialData::get_data(records)
+
+# Submit order for any new records not downloaded
+records <- getSpatialData::order_data(records)
+
+return(records)
+}
+
+#' @title extract and mask raster
+#' @description extracts raster files from compressed files, crops to Portal, 
+#' and masks with pixel_qa layer
+#'
+#' @param records new records to be used
+#' @param targetpath path where the cropped and masked rasters are to be stored
+#'
+extract_and_mask_raster <- function(records, targetpath = tempdir()) {
+  tarfile <- as.character(unlist(records["dataset_file"]))[1]
+  record_id <- as.character(unlist(records["record_id"]))[1]
+  # extract ndvi and pixel_qa rasters
+  untar(tarfile, exdir = targetpath)
+  # create portal box object for cropping
+  portal_area <- create_portal_area()
+  
+  # read in raster files; crop to portal_box; delete full-size raster files  
+  r <- raster::raster(paste0(targetpath,"/", record_id, "_pixel_qa.tif"))
+  scene <- raster::crop(r,portal_area)
+  
+  m <- raster::raster(paste0(targetpath,"/", record_id, "_sr_ndvi.tif"))
+  pixelqa <- raster::crop(m,portal_area)
+  
+  # mask ndvi data; write masked raster to file
+  s <- mask_landsat(record_id, scene, pixelqa)
+  return(s)
+}
+
+#' @description function to mask landsat scene
+#' 
+#' @param record_id new record to be used
+#' @param scene raster of scene to be masked
+#' @param pixelqa raster of pixel QA info
+#' 
+
+mask_landsat <- function(record_id, scene, pixelqa) {
+  
+  # which landsat satellite data is from determins what the "clear" values of pixel_qa are
+  # landsat8: from https://landsat.usgs.gov/sites/default/files/documents/lasrc_product_guide.pdf page 21
+  clearvalues = c(322, 386, 834, 898, 1346)
+  
+  pixelqa[!(pixelqa %in% clearvalues)] <- NA
+  s <- raster::mask(scene,pixelqa)
```

Comment:
Specifying arguments might be good practice for us here.
```suggestion
  s <- raster::mask(x=scene, mask=pixelqa)
```